usecases/case_usecase.go

```diff
@@ -762,94 +764,93 @@ func (usecase *CaseUseCase) CreateCaseFile(ctx context.Context, input models.Cre
-			return models.Case{}, err
-		}
-
-		newCaseFileId := uuid.NewString()
-		if err := usecase.repository.CreateDbCaseFile(
-			ctx,
-			tx,
-			models.CreateDbCaseFileInput{
-				Id:            newCaseFileId,
-				BucketName:    usecase.caseManagerBucketUrl,
-				CaseId:        input.CaseId,
-				FileName:      input.File.Filename,
-				FileReference: newFileReference,
-			},
-		); err != nil {
+	for _, fileHeader := range input.Files {
+		newFileReference := fmt.Sprintf("%s/%s/%s", creds.OrganizationId, input.CaseId, uuid.NewString())
+		writer, err := usecase.blobRepository.OpenStream(ctx, usecase.caseManagerBucketUrl, newFileReference, fileHeader.Filename)
+		if err != nil {
 			return models.Case{}, err
 		}
+		defer writer.Close() // We should still call Close when we are finished writing to check the error if any - this is a no-op if Close has already been called
 
-		resourceType := models.CaseFileResourceType
-		err = usecase.repository.CreateCaseEvent(ctx, tx, models.CreateCaseEventAttributes{
-			CaseId:         input.CaseId,
-			UserId:         userId,
-			EventType:      models.CaseFileAdded,
-			ResourceType:   &resourceType,
-			ResourceId:     &newCaseFileId,
-			AdditionalNote: &input.File.Filename,
-		})
+		file, err := fileHeader.Open()
 		if err != nil {
+			return models.Case{}, errors.Wrap(models.BadParameterError, err.Error())
+		}
+		if _, err := io.Copy(writer, file); err != nil {
 			return models.Case{}, err
 		}
-
-		updatedCase, err := usecase.getCaseWithDetails(ctx, tx, input.CaseId)
-		if err != nil {
+		if err := writer.Close(); err != nil {
 			return models.Case{}, err
 		}
 
-		err = usecase.webhookEventsUsecase.CreateWebhookEvent(ctx, tx, models.WebhookEventCreate{
-			Id:             webhookEventId,
-			OrganizationId: updatedCase.OrganizationId,
-			EventContent:   models.NewWebhookEventCaseFileCreated(updatedCase),
+		webhookEventId := uuid.New().String()
+
+		err = usecase.transactionFactory.Transaction(ctx, func(
+			tx repositories.Transaction,
+		) error {
+			if err := usecase.createCaseContributorIfNotExist(ctx, tx, input.CaseId, userId); err != nil {
+				return err
+			}
+
+			newCaseFileId := uuid.NewString()
+			if err := usecase.repository.CreateDbCaseFile(
+				ctx,
+				tx,
+				models.CreateDbCaseFileInput{
+					Id:            newCaseFileId,
+					BucketName:    usecase.caseManagerBucketUrl,
+					CaseId:        input.CaseId,
+					FileName:      fileHeader.Filename,
+					FileReference: newFileReference,
+				},
+			); err != nil {
+				return err
+			}
+
+			resourceType := models.CaseFileResourceType
+			err = usecase.repository.CreateCaseEvent(ctx, tx, models.CreateCaseEventAttributes{
+				CaseId:         input.CaseId,
+				UserId:         userId,
+				EventType:      models.CaseFileAdded,
+				ResourceType:   &resourceType,
+				ResourceId:     &newCaseFileId,
+				AdditionalNote: &fileHeader.Filename,
+			})
```

Comment:
Same question: should we keep one event per file, or an event for "all files"

Reply:
tonight I'd say one event for multiple files, to discuss tomorrow

Reply:
One case event per file, with the new history filtering this should'nt be an issue for now (activity log polution is mitigated)